bluesky/plan_stubs.py

```diff
@@ -773,33 +773,40 @@ def trigger_and_read(devices, name='primary'):
     msg : Msg
         messages to 'trigger', 'wait' and 'read'
     """
-    # If devices is empty, don't emit 'create'/'save' messages.
-    if not devices:
-        yield from null()
-    devices = separate_devices(devices)  # remove redundant entries
-    rewindable = all_safe_rewind(devices)  # if devices can be re-triggered
-
-    def inner_trigger_and_read():
-        grp = _short_uid('trigger')
-        no_wait = True
-        for obj in devices:
-            if hasattr(obj, 'trigger'):
-                no_wait = False
-                yield from trigger(obj, group=grp)
-        # Skip 'wait' if none of the devices implemented a trigger method.
-        if not no_wait:
-            yield from wait(group=grp)
-        yield from create(name)
-        ret = {}  # collect and return readings to give plan access to them
-        for obj in devices:
-            reading = (yield from read(obj))
-            if reading is not None:
-                ret.update(reading)
-        yield from save()
-        return ret
-    from .preprocessors import rewindable_wrapper
-    return (yield from rewindable_wrapper(inner_trigger_and_read(),
-                                          rewindable))
+
+    try:
+        # If devices is empty, don't emit 'create'/'save' messages.
+        if not devices:
+            yield from null()
+        devices = separate_devices(devices)  # remove redundant entries
+        rewindable = all_safe_rewind(devices)  # if devices can be re-triggered
+
+        def inner_trigger_and_read():
+            grp = _short_uid('trigger')
+            no_wait = True
+            for obj in devices:
+                if hasattr(obj, 'trigger'):
+                    no_wait = False
+                    yield from trigger(obj, group=grp)
+            # Skip 'wait' if none of the devices implemented a trigger method.
+            if not no_wait:
+                yield from wait(group=grp)
+            yield from create(name)
+            ret = {}  # collect and return readings to give plan access to them
+            for obj in devices:
+                reading = (yield from read(obj))
+                if reading is not None:
+                    ret.update(reading)
+            yield from save()
+            return ret
+        from .preprocessors import rewindable_wrapper
+        return (yield from rewindable_wrapper(inner_trigger_and_read(),
+                                              rewindable))
+
+    except IllegalMessageSequence as excep:
+        excep_str = ('While performing a "trigger_and_read" the following '
+                     'message sequence issue occured: ' + str(excep))
```

Comment:
Spelling fix, and also an exception's `repr` includes the type while the `str` just includes the message.

```suggestion
                     f'message sequence issue occurred: {excep!r}')
```